src/Lecture2.hs

```diff
@@ -260,11 +437,33 @@ data EvalError
     = VariableNotFound String
     deriving (Show, Eq)
 
+maybeHead :: [a] -> Maybe a
+maybeHead [] = Nothing
+maybeHead (x:_) = Just x
```

Comment:
That's a correct implementation! 👏🏻 

But you don't need to define such function manually. You can use [listToMaybe](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Maybe.html#v:listToMaybe) instead. I see that you already import the `Data.Maybe` module 🙂